bots/machine/machine_core/directories.py

```diff
@@ -18,14 +18,30 @@
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
 import os
+import subprocess
 
-from .constants import BOTS_DIR
+from .constants import BOTS_DIR, GIT_DIR
+
+def get_git_config(variable):
+    if not os.path.exists(GIT_DIR):
+        return None
+
+    try:
+        myenv = os.environ.copy()
+        myenv["GIT_DIR"] = GIT_DIR
+        return subprocess.check_output(["git", "config", variable], text=True, env=myenv).strip()
```

Comment:
`text` is not a valid kwarg for [subprocess.check_output()](https://docs.python.org/3.4/library/subprocess.html). `universal_newlines=True` does that.

Reply:
Quoting the docs for subprocess.check_output() (emphasis mine):

> The arguments shown above are merely the most common ones, described below in Frequently Used Arguments (hence the use of keyword-only notation in the abbreviated signature). **The full function signature is largely the same as that of the Popen constructor - this functions passes all supplied arguments other than input and timeout directly through to that interface.** In addition, stdout is not permitted as an argument, as it is used internally to collect the output from the subprocess.

Reply:
Actually, it seems like we're looking at a different version of the docs.  https://docs.python.org/3/library/subprocess.html#subprocess.check_output shows the kwarg `text=`.

Reply:
Right, I deliberately didn't look at the very latest, as it's useful to also keep this working on e. g. Fedora 29.